cpplint_unittest.py

```diff
@@ -604,6 +604,86 @@ def testErrorSuppression(self):
+                             '// NOLINTBEGIN',
+                             'long a = (int64) 65;'
+                             'long a = 65;',
+                             '// NOLINTEND'
+                             '//  ./command' + (' -verbose' * 80),
+                             ''],
+                            error_collector)
+    self.assertEquals('Lines should be <= 80 characters long  '
+                      '[whitespace/line_length] [2]', error_collector.Results())
+    # NOLINTBEGIN(category) silences category warnings after it
+    error_collector = ErrorCollector(self.assert_)
+    cpplint.ProcessFileData('test.cc', 'cc',
+                            ['// Copyright 2014 Your Company.',
+                             '// NOLINTBEGIN(readability/casting,runtime/int)',
+                             'long a = (int64) 65;',
+                             'long a = 65;',
+                             '//  ./command' + (' -verbose' * 80),
+                             ''],
+                            error_collector)
+    self.assertEquals('Lines should be <= 80 characters long  '
+                      '[whitespace/line_length] [2]',
+                      error_collector.Results())
+    # NOLINTEND(category) will generate an error that categories are not supported
+    error_collector = ErrorCollector(self.assert_)
+    cpplint.ProcessFileData('test.cc', 'cc',
+                            ['// Copyright 2014 Your Company.',
+                             '// NOLINTBEGIN(readability/casting,runtime/int)',
+                             'long a = (int64) 65;',
+                             'long a = 65;',
+                             '// NOLINTEND(readability/casting)',
+                             ''],
+                            error_collector)
+    self.assertEquals('NOLINT categories not supported in block END: readability/casting  '
+                      '[readability/nolint] [5]',
+                      error_collector.Results())
+    # nested NOLINTBEGIN is not allowed
+    error_collector = ErrorCollector(self.assert_)
+    cpplint.ProcessFileData('test.cc', 'cc',
+                            ['// Copyright 2014 Your Company.',
+                             '// NOLINTBEGIN(readability/casting,runtime/int)',
+                             'long a = (int64) 65;',
+                             '// NOLINTBEGIN(runtime/int)',
+                             'long a = 65;',
+                             '// NOLINTEND(*)',
+                             ''],
+                            error_collector)
+    self.assertEquals('NONLINT block already defined on line 2  '
+                      '[readability/nolint] [5]', error_collector.Results())
+    # error if unmatched NOLINTEND
+    self.TestLint(
+        '// NOLINTEND',
+        'Not in a NOLINT block  '
+        '[readability/nolint] [5]')
+    self.TestLint(
+        '// NOLINTEND(*)',
+        'Not in a NOLINT block  '
+        '[readability/nolint] [5]')
```

Comment:
Nice testing. I currently wonder if it should be an error to open a block but not close it. This could indicate mistakenly having deleted only a NOLINTEND but not the NOLINTBEGIN. Did you consider that case?

Reply:
I thought about it but didn't feel it should be required but it makes sense that it should be similar to blocks of code in programming. I will add that.